js/assessment.js

```diff
@@ -1,409 +1,394 @@
+    _.defer(() => {
+      Adapt.trigger('assessment:complete', this.getState());
+    });
+
+    return true;
+  },
+
+  _setupSingleAssessmentConfiguration(assessmentState) {
+    const assessmentsConfig = Adapt.course.get('_assessment');
+    $.extend(true, assessmentsConfig, {
+      _isPercentageBased: assessmentState.isPercentageBased,
+      _scoreToPass: assessmentState.scoreToPass
+    });
+    Adapt.course.set('_assessment', assessmentsConfig);
+  },
+
+  _getAssessmentByPageId(pageId) {
+    return this._assessments._byPageId[pageId];
+  },
+
+  _getStateByAssessmentId(assessmentId) {
+    if (assessmentId === undefined) {
+      return null;
+    }
 
-    _setPageProgress: function() {
-      // set _subProgressTotal and _subProgressComplete on pages that have assessment progress indicator requirements
+    return this._assessments._byAssessmentId[assessmentId].getState();
+  },
 
-      for (var k in this._assessments._byPageId) {
+  _getStatesByAssessmentId() {
+    const states = {};
+    for (const assessmentModel of this._assessments) {
+      if (!assessmentModel.get('_isAvailable')) continue;
+      const state = assessmentModel.getState();
+      states[state.id] = state;
+    }
+    return states;
+  },
 
-        var assessments = this._assessments._byPageId[k];
+  _setPageProgress() {
+    // set _subProgressTotal and _subProgressComplete on pages that have assessment progress indicator requirements
 
-        var assessmentsTotal = assessments.length;
-        var assessmentsPassed = 0;
+    for (const k in this._assessments._byPageId) {
 
-        for (var i = 0, l = assessments.length; i < l; i++) {
-          var assessmentState = assessments[i].getState();
+      const assessments = this._assessments._byPageId[k];
 
-          if (assessmentState.includeInTotalScore && !assessmentState.isPass) continue;
+      const assessmentsTotal = assessments.length;
+      let assessmentsPassed = 0;
 
-          if (assessmentState.isComplete) {
-            assessmentsPassed++;
-          }
-        }
+      for (const assessment of assessments) {
+        const assessmentState = assessment.getState();
 
-        try {
-          var pageModel = Adapt.findById(k);
-          pageModel.set({
-            _subProgressTotal: assessmentsTotal,
-            _subProgressComplete: assessmentsPassed
-          });
-        } catch (e) {
+        if (assessmentState.includeInTotalScore && !assessmentState.isPass) continue;
 
+        if (assessmentState.isComplete) {
+          assessmentsPassed++;
         }
-
       }
-    },
 
-    _addToAssessmentIdMap: function(id, model) {
-      if (id === undefined) {
-        Adapt.log.warn('An assessment has been registered with an undefined value for "_id"');
-        return;
-      }
-
-      if (id === '') {
-        Adapt.log.warn('An assessment has been registered with an empty value for "_id"');
-      }
+      try {
+        const pageModel = Adapt.findById(k);
+        pageModel.set({
```

Comment:
No longer require _try...catch_.
```suggestion
        pageModel?.set({
```